src/common/convert.c

```diff
@@ -58,8 +59,19 @@ struct BufferConvertContextSource {
     size_t pos;
 };
 
+// Number of blocks transferred between mainloop pumps
+#define PUMP_MAINLOOP_EVERY_X_BLOCKS 300
+
 static size_t convert_io_transfer(struct ConvertIO *io, char *buffer, size_t len)
 {
+    static int iterations = 0;
+    if (iterations++ >= PUMP_MAINLOOP_EVERY_X_BLOCKS) {
```

Comment:
Nice find -- doesn't matter too much in practice (off-by-one-in-300, where 300 is just a wild guess between not-starving-the-mainloop-and-not-slowing-down-the-transfer), but better do what the comment and the variable name says ;)